### ISSUE TITLE
[RFC] Support more tools natively

### ISSUE THREAD

**Issue by chshersh** [chshersh/tool-sync#85]
Blocked by #84 

Currently `tool-sync` supports only a handful of tools. But let's add more useful tools! Feel free to add ideas if the tool follows [suggestions for adding a new tool](https://github.com/chshersh/tool-sync#adding-a-new-tool).

At this moment, I see the following tools as good additions:

- [ ] [procs](https://github.com/dalance/procs)
- [ ] [sd](https://github.com/chmln/sd)
- [ ] [dust](https://github.com/bootandy/dust)
- [ ] [hyperfine](https://github.com/sharkdp/hyperfine)
- [ ] [bottom](https://github.com/ClementTsang/bottom)
- [ ] [tealdeer](https://github.com/dbrgn/tealdeer)
- [ ] [bandwhich](https://github.com/imsnif/bandwhich)
- [ ] [zoxide](https://github.com/ajeetdsouza/zoxide)
- [ ] [shellcheck](https://github.com/koalaman/shellcheck)
- [ ] [gh](https://github.com/cli/cli)
- [ ] [hlint](https://github.com/ndmitchell/hlint)

I would love to add all the tools from the above list to `tool-sync`'s native support (after #84 is done) as I use or want to start using them. Until this issue is done, feel free to suggest other great tools 👏🏻

**Comment by hdhoang** [comment]
this is unblocked now, i'll give shellcheck & k9s a try. sc uses tar.xz, which needs support in archive module too.

i wanted to start out with nushell, but it has some companion binaries of pattern `nu_*`. perhaps exe_name can be a glob for this case?

**Comment by chshersh** [comment]
> this is unblocked now

Thanks! I'll update the description ✅ 

> sc uses tar.xz, which needs support in archive module too.

Indeed, this needs to be supported separately. Feel free to create a separate issue for supporting the `xz` format. All the relevant logic is currently written in this module:

* https://github.com/chshersh/tool-sync/blob/main/src/sync/archive.rs

> i wanted to start out with nushell, but it has some companion binaries of pattern nu_*. perhaps exe_name can be a glob for this case?

Could you elaborate a bit more on this? `nushell` is not on the list above so I haven't looked into it. Are you saying that to _sync / install_ `nushell` you need to **copy** several binary files? If that's the case, then indeed, `tool-sync` currently copies only a single file. I propose to create another separate issue to discuss how to sync multiple files from the asset.
